### PR TITLE
Add support for nested unions / intersections

### PR DESCRIPTION
**Pull request by mwhansen**

### CODE REVIEW

**Comment by mwhansen** [comment]
@chaten please review and +1

735e3eb is the most relevant commit for the actual changes

File: src/django_delayed_union/intersection.py [hunk 1]
```diff
@@ -3,7 +3,17 @@
 
 class DelayedIntersectionQuerySet(DelayedQuerySet):
     def __init__(self, *querysets):
-        return super(DelayedIntersectionQuerySet, self).__init__(*querysets)
+        # Handle the case when a DelayedUnionQuerySet is passed in
```
Comment (chaten) [inline src/django_delayed_union/intersection.py]:
```suggestion
        # Handle the case when a DelayedIntersectionQuerySet is passed in
```

File: src/django_delayed_union/union.py [hunk 2]
```diff
@@ -11,7 +11,21 @@ def __init__(self, *querysets, **kwargs):
                     unexpected_kwarg
                 )
             )
-        return super(DelayedUnionQuerySet, self).__init__(*querysets, **kwargs)
+
+        # Handle the case when a DelayedUnionQuerySet is passed in
+        expanded_querysets = []
+        for queryset in querysets:
+            if isinstance(queryset, DelayedUnionQuerySet):
```
Comment (chaten) [inline src/django_delayed_union/union.py:18]: These two changes to the constructor feel the same.

Would it be a good idea to push these down to DelayedQuerySet to do the expanding?

Something like
```
if type(self) is type(queryset):
   # validate kwargs
   self._querysets.extend(queryset._querysets)
else:
   self._querysets.append(queryset)
end

Comment (mwhansen) [reply src/django_delayed_union/union.py:18]: It doesn't work for `.difference` so I was okay with the explicit duplication as opposed to adding  an abstraction.

**Review by chaten — APPROVED** [review]

**Pull request merged**